--- src/db/users.rs
+++ src/db/users.rs
@@ -19,10 +19,11 @@
 pub struct PublicUser {
     pub id: Uuid,
     pub user_name: String,
+    pub avatar_hash: Option<String>,
     pub created_at: sea_orm::entity::prelude::DateTimeWithTimeZone,
 }
 
 #[derive(Clone, Debug)]
 pub struct NewUser {
     pub user_name: String,
     pub display_name: String,
@@ -46,13 +47,15 @@
     created_at: sea_orm::entity::prelude::DateTimeWithTimeZone,
     friends_created_at: sea_orm::entity::prelude::DateTimeWithTimeZone,
+    use_gravatar: bool,
+    email: String,
 }
 
 impl From<user::Model> for PublicUser {
     fn from(value: user::Model) -> Self {
         Self {
             id: value.id,
             user_name: value.user_name,
+            avatar_hash: if value.use_gravatar { Some(format!("{:x}", md5::compute(value.email.to_lowercase()))) } else { None },
             created_at: value.created_at,
         }
     }
@@ -282,14 +285,15 @@
         db: &DatabaseConnection,
         user_id: Uuid,
     ) -> Result<Vec<super::friends::FriendEntry>, sea_orm::DbErr> {
         let rows = FriendEntryRow::find_by_statement(Statement::from_sql_and_values(
             DbBackend::Postgres,
-            r#"SELECT users.id, user_name, users.created_at, tmp.created_at_ret AS friends_created_at
+            r#"SELECT users.id, user_name, users.use_gravatar, users.email, users.created_at, tmp.created_at_ret AS friends_created_at
                FROM users INNER JOIN (SELECT * FROM get_friend_entries($1)) AS tmp ON users.id = tmp.friend_user_id_ret
                ORDER BY user_name"#,
             vec![user_id.into()],
         ))
         .all(db)
         .await?;
 
         Ok(rows
             .into_iter()
             .map(|row| super::friends::FriendEntry {
                 created_at: row.friends_created_at,
                 friend: PublicUser {
                     id: row.id,
                     user_name: row.user_name,
+                    avatar_hash: if row.use_gravatar { Some(format!("{:x}", md5::compute(row.email.to_lowercase()))) } else { None },
                     created_at: row.created_at,
                 },
             })
             .collect())
